src/resources.py

```diff
@@ -119,11 +132,19 @@ def list_resources(label, label_value, target_folder, request_url, request_metho
 
 
 def _process_secret(dest_folder, secret, resource, unique_filenames, enable_5xx, is_removed=False):
+    files_changed = False
+
+    old_secret = _resources_object_map.get(secret.metadata.namespace + secret.metadata.name) or copy.deepcopy(secret)
+    if is_removed:
+        _resources_object_map.pop(secret.metadata.namespace + secret.metadata.name, None)
```

Comment:
Since list_resources is constantly being added to the cache, you need to pop objects that are not in the list, so I pop the objects out of the cache on line 120.

For compatibility with `_watch_resource_iterator`, the one here will perform one more pop operation, so I set the second parameter to None.

Reply:
Sure, but if passing over `True` to `_process_secret`/`_process_config_map` already deletes to the cache being cleared, there's no point in doing it on line 120 already, correct? `list_resources` can just rely on the called methods to remove deleted objects from the cache, can't it?

Reply:
> Sure, but if passing over `True` to `_process_secret`/`_process_config_map` already deletes to the cache being cleared, there's no point in doing it on line 120 already, correct? `list_resources` can just rely on the called methods to remove deleted objects from the cache, can't it?

I get it